tool/travis.sh

```diff
@@ -32,7 +32,7 @@ while (( "$#" )); do
   test) echo
     echo -e '\033[1mTASK: test\033[22m'
     echo -e 'pub run test --run-skipped'
-    pub run test --run-skipped || EXIT_CODE=$?
+    pub run test --timeout 2m --run-skipped -j 1 || EXIT_CODE=$?
```

Comment:
Puzzling as it is, in the end `concurrency=1` solved the flaky tests. Bummer.
I'm keeping the rest of the fixes and merging after I've validated it in staging.
/cc @sortie

Reply:
https://github.com/dart-lang/pub-dartlang-dart/issues/1437